src/command_modules/azure-cli-role/azure/cli/command_modules/role/_params.py

```diff
@@ -127,9 +127,11 @@ def load_arguments(self, _):
     with self.argument_context('ad user') as c:
         c.argument('mail_nickname', help='mail alias. Defaults to user principal name')
         c.argument('force_change_password_next_login', arg_type=get_three_state_flag())
+        c.argument('account_enabled', arg_type=get_three_state_flag(), help='enable the user account')
+        c.argument('password', help='user password')
+        c.argument('upn_or_object_id', help='The object ID or principal name of the user for which to get information')
```

Comment:
Is there a better name for this argument?  Why not simply have `options_list=['--upn', '--object-id']`... or break it out into two arguments with distinct help text? My gut tells me we shouldn't have "or" in our argument names.

Reply:
But I can see the sdk function is using that as a parameter in the code below.
client.update(upn_or_object_id=upn_or_object_id, parameters=update_parameters)

Reply:
This is the named used by SDK, but we should not have exposed it through `az ad user show`. Since the name has been out there for 2 years, breaking it right now might be a stretch. For better usability, I can add `--id` as another option, this will get us the consistency across `app`, `sp`, and `user`. Also `upn` is a logical id for a user, so `--id` is a right term.

Reply:
We can add the split options and deprecate the one with `or` in it.